File: cpp/724.cpp
#include<iostream>
#include<vector>

using namespace std;

class Solution {
public:
    int pivotIndex(vector<int>& nums) {
        int leftSum = 0, rightSum = 0;
        for(int i = 0; i < nums.size(); i++)
            rightSum += nums[i];
        for(int i = 0; i < nums.size(); i++){
            rightSum -= nums[i];
            if(leftSum == rightSum) return i;
            leftSum += nums[i];
        }
        return -1;
    }
};

int main() {
    Solution s;
    int n;
    cin >> n;
    vector<int> nums(n);
    for(int i = 0; i < n; i++)
        cin >> nums[i];
    cout << s.pivotIndex(nums) << endl;
    return 0;
}

File: python/724.py
class Solution:
    def pivotIndex(self, nums: list[int]) -> int:
        leftSum, rightSum = 0, sum(nums)
        for i, num in enumerate(nums):
            rightSum -= num
            if leftSum == rightSum:
                return i
            leftSum += num
        return -1
    
if __name__ == '__main__':
    s = Solution()
    nums = list(map(int, input().split()))
    print(s.pivotIndex(nums))